OPHD/States/MapViewStateIO.cpp

```diff
@@ -145,22 +146,20 @@ void MapViewState::load(const std::string& filePath)
 	mTileMap = nullptr;
 
 	XmlElement* map = root->firstChildElement("properties");
-	int depth = 0;
-	std::string sitemap;
-	std::string tilesetPath;
 	XmlAttribute* attribute = map->firstAttribute();
 	while (attribute)
 	{
-		if (attribute->name() == "diggingdepth") { attribute->queryIntValue(depth); }
-		else if (attribute->name() == "sitemap") { sitemap = attribute->value(); }
-		else if (attribute->name() == "tset") { tilesetPath = attribute->value(); }
+		if (attribute->name() == "diggingdepth") { attribute->queryIntValue(mPlanetAttributes.maxDepth); }
+		else if (attribute->name() == "sitemap") { mPlanetAttributes.mapImagePath = attribute->value(); }
+		else if (attribute->name() == "tset") { mPlanetAttributes.tilesetPath = attribute->value(); }
+		else if (attribute->name() == "meansolardistance") { mPlanetAttributes.meanSolarDistance = std::stof(attribute->value()); }
```

Comment:
This seems a bit inconsistent with how float values are read. Is there a [`queryFloatValue`](https://leethomason.github.io/tinyxml2/classtinyxml2_1_1_x_m_l_attribute.html#a049dea6449a6259b6cfed44a9427b607) in TinyXML 1?

Not really sure if one way should be preferred over the other.

Reply:
I appear to only have a query double or int function available via NAS2D. Opened an issue on NAS2D to request feature.